### DISCUSSION TITLE
get end of quarter?

### DISCUSSION THREAD

**Discussion by J4v4Scr1pt** [adobe/react-spectrum#7039] [Q&A] [answered]
Hi!

I currently trying to replace our dependencie on dayJs and replace it with internationalized/date.
But I'm stuck on how to get the end of current quarter based on selected date?

I have searched the docs and looked at the code, but can't find anything that would help finding a solution to this.

Any ideas would be much appreciated!

**Comment by reidbarber** [comment, marked as answer]
This isn't built into the library, but you should be able to create a utility function like this:

```ts
import {CalendarDate, today, getLocalTimeZone} from '@internationalized/date'

function getEndOfQuarter(date: CalendarDate) {
  const currentQuarter = Math.floor((date.month - 1) / 3) + 1;
  const lastMonthOfQuarter = currentQuarter * 3;
  const firstDayOfLastMonth = new CalendarDate(date.year, lastMonthOfQuarter, 1);
  const firstDayOfNextMonth = firstDayOfLastMonth.add({months: 1});
  const lastDayOfQuarter = firstDayOfNextMonth.subtract({days: 1});
  return lastDayOfQuarter;
}

const endOfQuarter = getEndOfQuarter(today(getLocalTimeZone()));
```

**Reply by J4v4Scr1pt** [reply]
Thx m8 🚀!

I also needed startOfQuarter, created this util file `quarter.ts`

```
import { CalendarDate } from '@internationalized/date';

export function endOfQuarter(date: CalendarDate) {
	const currentQuarter = Math.floor((date.month - 1) / 3) + 1;
	const lastMonthOfQuarter = currentQuarter * 3;
	const firstDayOfLastMonth = new CalendarDate(date.year, lastMonthOfQuarter, 1);
	const firstDayOfNextMonth = firstDayOfLastMonth.add({ months: 1 });
	const lastDayOfQuarter = firstDayOfNextMonth.subtract({ days: 1 });
	return lastDayOfQuarter;
}

export function startOfQuarter(date: CalendarDate) {
	const currentQuarter = Math.floor((date.month - 1) / 3) + 1;
	const lastMonthOfQuarter = currentQuarter * 3;
	const firstDayOfLastMonth = new CalendarDate(date.year, lastMonthOfQuarter, 1);
	const firstDayOfFirstMonth = firstDayOfLastMonth.subtract({ months: 2 });
	return firstDayOfFirstMonth;
}

```

How come these are not built in?

**Reply by reidbarber** [reply]
I think it's a bit application-specific to be included in the library IMO.

**Reply by J4v4Scr1pt** [reply]
IMO I think the opposite ofc 😅. Mostly because of [this statement](https://github.com/adobe/react-spectrum/discussions/2775).
If it should be a replacement for e.g moment. This kind of functionality should be included. 
But I'm no maintainer and I'm ok with creating a few util functions. 😄. Thx again for your help 🙏 !
